### PR TITLE
More Cosmos Pylint work

### PR DESCRIPTION
**Pull request by bryevdv**
This is a follow-on to #6851 These Pylint issues were fixed in this PR:

```

  64  C0301  Line too long (126/120) (line-too-long)
  23  W0622  Redefining built-in 'id' (redefined-builtin)
  24  C1801  Do not use `len(SEQUENCE)` to determine if a sequence is empty (len-as-condition)
  13  C0411  standard import "import base64" should be placed before "import six" (wrong-import-order)
   4  R1710  Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
   4  C0123  Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)
   3  C0200  Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
   1  R1720  Unnecessary "else" after "raise" (no-else-raise)
```

Additionally `black -l 120`  was re-run to keep everything consistent. 

These are the current remaining pylint issues:

```
  56  W0212  Access to a protected member _useMultipleWriteLocations of a client class (protected-access)
  15  R0201  Method could be a function (no-self-use)
  11  C0301  Line too long (126/120) (line-too-long)
   6  W0201  Attribute '_database_link' defined outside __init__ (attribute-defined-outside-init)
   4  R0902  Too many instance attributes (11/10) (too-many-instance-attributes)
   3  R0911  Too many return statements (7/6) (too-many-return-statements)
   3  W0231  __init__ method from base class 'dict' is not called (super-init-not-called)
   2  R0904  Too many public methods (86/20) (too-many-public-methods)
   2  W0223  Method '_fetch_next_block' is abstract in class '_QueryExecutionContextBase' but is not overridden (abstract-method)
   2  W0703  Catching too general exception Exception (broad-except)
   1  E1101  Instance of 'Exception' has no 'message' member (no-member)
   1  R0912  Too many branches (40/20) (too-many-branches)
   1  R0915  Too many statements (89/50) (too-many-statements)
   1  R1702  Too many nested blocks (6/5) (too-many-nested-blocks)
```

The remaining `too-long-line` are long type comments, I am not sure (offhand) the best way to handle those. I think there is a "per-argument" syntax that could be used.

Most of the other remaining issues are things I think the Cosmos team might want to weigh in on. 

I would recommend again merging this PR when it is green and tacking another chunk in a separate PR. 

cc @srinathnarayanan @annatisch

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/cosmos/azure-cosmos/azure/cosmos/_base.py [hunk 1]
```diff
@@ -334,7 +335,7 @@ def IsNameBased(link):
     parts = link.split("/")
 
     # First part should be "dbs"
-    if len(parts) == 0 or not parts[0] or not parts[0].lower() == "dbs":
+    if not (parts and parts[0].lower() == "dbs"):
```
Comment (bryevdv) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_base.py]: After transforming `len(parts) == 0` to `not parts` the clause has the form `not X or not Y or not Z` which is equivalent to `not (X and Y and Z)` (deMorgan's law). Additionally, the check for `parts[0]` is not needed -- if parts has any elements, it has `parts[0]` and `parts[0]` is a string (output of `split`)

File: sdk/cosmos/azure-cosmos/azure/cosmos/_default_retry_policy.py [hunk 2]
```diff
@@ -54,11 +54,12 @@ def __init__(self, *args):
 
     def needsRetry(self, error_code):
         if error_code in DefaultRetryPolicy.CONNECTION_ERROR_CODES:
-            if len(self.args) > 0:
+            if self.args:
                 if (self.args[4]["method"] == "GET") or (http_constants.HttpHeaders.IsQuery in self.args[4]["headers"]):
                     return True
                 return False
             return True
+        return False
```
Comment (bryevdv) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_default_retry_policy.py]: Technically speaking this function returned `None` at this point before, but I believe `False` matches the intent

File: sdk/cosmos/azure-cosmos/azure/cosmos/_location_cache.py [hunk 3]
```diff
@@ -132,9 +132,7 @@ def resolve_service_endpoint(self, request):
         return endpoints[location_index % len(endpoints)]
 
     def should_refresh_endpoints(self):
-        most_preferred_location = (
-            self.preferred_locations[0] if (self.preferred_locations and len(self.preferred_locations) > 0) else None
-        )
+        most_preferred_location = self.preferred_locations[0] if self.preferred_locations else None
```
Comment (bryevdv) [inline sdk/cosmos/azure-cosmos/azure/cosmos/_location_cache.py]: `self.preferred_locations and len(self.preferred_locations) > 0` is redundant

**Comment by bryevdv** [comment]
Merging this now, will submit final PR for remaining pylint issues.

**Pull request merged**